server/src/main/java/org/cloudfoundry/identity/uaa/util/VmidentityUtils.java

```diff
@@ -35,15 +38,31 @@
 
     public static String getSystemDomain(String tenant, CasIdmClient idmClient) throws Exception {
         Collection<IIdentityStoreData> provider = idmClient.getProviders(tenant, EnumSet.of(DomainType.SYSTEM_DOMAIN));
-        // todo: should get security domain name, not provider name.
```

Comment:
let's keep this todo (technically we should return domain name, right now for system domain they are always the same)....
and submit....